src/megaapi_impl.cpp

```diff
@@ -18942,21 +18923,19 @@ void MegaApiImpl::sendPendingRequests()
                     string localpath;
                     fsAccess->path2local(&path, &localpath);
 
-                    FileAccess *f = fsAccess->newfileaccess();
+                    auto f = fsAccess->newfileaccess();
                     if (!f->fopen(&localpath, 1, 0))
                     {
-                        delete f;
                         e = API_EREAD;
                         break;
                     }
 
                     if (!f->fread(&attrvalue, unsigned(f->size), 0, 0))
                     {
-                        delete f;
                         e = API_EREAD;
                         break;
                     }
-                    delete f;
+                    f.reset();
```

Comment:
this one closes the handle before we call putua(), which might call the application back, so it may be necessary - safer to do exactly what it did before in this case, I think.